Sources/core/objects/Human.cpp

```diff
@@ -82,6 +84,274 @@ bool Human::TryMove(Dir direct)
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_)
+        {
+            if (!CanTouch(pulled_object_))
```

Comment:
I don't want to move the pulled object if it's in your touch range.  I was thinking about storing the position of the human but the case when you are in the corners you cannot do diagonal `TryMove`, and you cannot do 2 TryMove because the timer, and even there is the problem that you don't know which way isn't blocked like you can got top-left and left-top you will be at same position but maybe the first doesn't work but the 2nd one will.